src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/handler/ValidateAnomalyDetectorActionHandler.java

```diff
@@ -0,0 +1,590 @@
+import org.elasticsearch.common.xcontent.NamedXContentRegistry;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.index.query.BoolQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.index.query.RangeQueryBuilder;
+import org.elasticsearch.rest.BytesRestResponse;
+import org.elasticsearch.rest.RestChannel;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.search.aggregations.AggregationBuilders;
+import org.elasticsearch.search.aggregations.metrics.Max;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+import org.elasticsearch.search.sort.FieldSortBuilder;
+import org.elasticsearch.search.sort.SortOrder;
+
+import com.amazon.opendistroforelasticsearch.ad.indices.AnomalyDetectionIndices;
+import com.amazon.opendistroforelasticsearch.ad.model.AnomalyDetector;
+import com.amazon.opendistroforelasticsearch.ad.model.Feature;
+import com.amazon.opendistroforelasticsearch.ad.model.IntervalTimeConfiguration;
+import com.amazon.opendistroforelasticsearch.ad.model.ValidateResponse;
+import com.amazon.opendistroforelasticsearch.ad.util.ParseUtils;
+import com.amazon.opendistroforelasticsearch.ad.util.RestHandlerUtils;
+
+/**
+ * Anomaly detector REST action handler to process POST request.
+ * POST request is for validating anomaly detector.
+ */
+public class ValidateAnomalyDetectorActionHandler extends AbstractActionHandler {
+
+    protected static final String AGG_NAME_MAX = "max_timefield";
+    protected static final int NUM_OF_INTERVAL_SAMPLES = 128;
+    protected static final int MAX_NUM_OF_SAMPLES_VIEWED = 128;
+    protected static final int NUM_OF_INTERVALS_CHECKED = 256;
+    protected static final double SAMPLE_SUCCESS_RATE = 0.75;
+    protected static final int FEATURE_VALIDATION_TIME_BACK_MINUTES = 10080;
+    protected static final int NUM_OF_INTERVALS_CHECKED_FILTER = 384;
+    protected static final long MAX_INTERVAL_LENGTH = 2592000000L;
+    protected static final long HISTORICAL_CHECK_IN_MS = 7776000000L;
+    protected static final String NAME_REGEX = "[a-zA-Z0-9._-]+";
+    protected static final double INTERVAL_RECOMMENDATION_MULTIPLIER = 1.2;
+
+    private final AnomalyDetectionIndices anomalyDetectionIndices;
+    private final AnomalyDetector anomalyDetector;
+
+    private final Logger logger = LogManager.getLogger(IndexAnomalyDetectorActionHandler.class);
+    private final Integer maxAnomalyDetectors;
+    private final Integer maxAnomalyFeatures;
+    private final TimeValue requestTimeout;
+    private final NamedXContentRegistry xContent;
+
+    private ValidateResponse responseValidate;
+    private Map<String, List<String>> failuresMap;
+    private Map<String, List<String>> suggestedChangesMap;
+    private Boolean inferringInterval;
+    private AtomicBoolean inferAgain;
```

Comment:
Can you try making all of them final? I guess code can work even they are `final` since they are `object`

Reply:
fixed